examples/transfer_from_omnilock.rs

```diff
@@ -0,0 +1,375 @@
+            // Send transaction
+            let tx_info: TxInfo = serde_json::from_slice(&fs::read(&tx_file)?)?;
+            println!("> tx: {}", serde_json::to_string_pretty(&tx_info.tx)?);
+            let outputs_validator = Some(json_types::OutputsValidator::Passthrough);
+            let _tx_hash = CkbRpcClient::new(ckb_rpc.as_str())
+                .send_transaction(tx_info.tx.inner, outputs_validator)
+                .expect("send transaction");
+            println!(">>> tx sent! <<<");
+        }
+    }
+
+    Ok(())
+}
+
+fn build_omnilock_addr(args: &BuildOmniLockAddrArgs) -> Result<(), Box<dyn StdErr>> {
+    let mut ckb_client = CkbRpcClient::new(args.ckb_rpc.as_str());
+    let cell =
+        build_omnilock_cell_dep(&mut ckb_client, &args.omnilock_tx_hash, args.omnilock_index)?;
+    let config = OmniLockConfig::new_pubkey_hash_with_lockarg(args.receiver.payload().args());
+    let address_payload = {
+        let args = config.build_args();
+        ckb_sdk::AddressPayload::new_full(ScriptHashType::Type, cell.type_hash.pack(), args)
+    };
+    let lock_script = Script::from(&address_payload);
+    let resp = serde_json::json!({
+        "mainnet": Address::new(NetworkType::Mainnet, address_payload.clone(), true).to_string(),
+        "testnet": Address::new(NetworkType::Testnet, address_payload.clone(), true).to_string(),
+        "lock-arg": format!("0x{}", hex_string(address_payload.args().as_ref())),
+        "lock-hash": format!("{:#x}", lock_script.calc_script_hash())
+    });
+    println!("{}", serde_json::to_string_pretty(&resp)?);
+    Ok(())
+}
+
+fn gen_omnilock_tx(args: &GenTxArgs) -> Result<(), Box<dyn StdErr>> {
+    let (tx, omnilock_config) = build_transfer_tx(args)?;
+    let tx_info = TxInfo {
+        tx: json_types::TransactionView::from(tx),
+        omnilock_config,
+    };
+    fs::write(&args.tx_file, serde_json::to_string_pretty(&tx_info)?)?;
+    Ok(())
+}
+
+fn build_transfer_tx(
+    args: &GenTxArgs,
+) -> Result<(TransactionView, OmniLockConfig), Box<dyn StdErr>> {
+    let sender_key = secp256k1::SecretKey::from_slice(args.sender_key.as_bytes())
+        .map_err(|err| format!("invalid sender secret key: {}", err))?;
+    let pubkey = secp256k1::PublicKey::from_secret_key(&SECP256K1, &sender_key);
+    let mut ckb_client = CkbRpcClient::new(args.ckb_rpc.as_str());
+    let cell =
+        build_omnilock_cell_dep(&mut ckb_client, &args.omnilock_tx_hash, args.omnilock_index)?;
+    let omnilock_config = OmniLockConfig::new_pubkey_hash(&pubkey.into());
+    // Build CapacityBalancer
+    let sender = Script::new_builder()
+        .code_hash(cell.type_hash.pack())
+        .hash_type(ScriptHashType::Type.into())
+        .args(omnilock_config.build_args().pack())
+        .build();
+    let placeholder_witness = omnilock_config.placeholder_witness();
+    // set fee_rate to 2000 to reserve for secp256k1_data_dep
+    let balancer = CapacityBalancer::new_simple(sender, placeholder_witness, 2000);
```

Comment:
You should use `fee_rate=1000` to verify the fee calculation is right.

Reply:
It's just a workaround for adjust the transaction fee after add additional secp256k1 data dependency.

After insert a secp256k1_data dependency cell, more fee will be need, 1000 can only give minimal fee, the transaction will be reject.